ergo-lib/src/chain/address.rs

```diff
@@ -265,24 +310,44 @@ impl AddressEncoder {
     /// parse address from Base58 encoded string
     pub fn parse_address_from_str(&self, str: &str) -> Result<Address, AddressEncoderError> {
         let bytes = bs58::decode(str).into_vec()?;
-        if bytes.len() < AddressEncoder::MIN_ADDRESS_LENGTH {
+        if bytes.len() < 1 {
             return Err(AddressEncoderError::InvalidSize);
         };
         self.check_head_byte(bytes[0])?;
-        AddressEncoder::try_parse_address(&bytes)
+        AddressEncoder::unchecked_parse_address_from_bytes(&bytes)
     }
 
-    /// parse address from Base58 encoded string
-    pub fn unchecked_parse_address_from_str(str: &str) -> Result<Address, AddressEncoderError> {
+    /// parse network+address from Base58 encoded string
+    pub fn unchecked_parse_network_address_from_str(str: &str) -> Result<NetworkAddress, AddressEncoderError> {
         let bytes = bs58::decode(str).into_vec()?;
-        if bytes.len() < AddressEncoder::MIN_ADDRESS_LENGTH {
+        AddressEncoder::unchecked_parse_network_address_from_bytes(&bytes)
+    }
+
+    /// parse network+address from Base58 encoded string
+    pub fn unchecked_parse_network_address_from_bytes(bytes: &[u8]) -> Result<NetworkAddress, AddressEncoderError> {
+        if bytes.len() < 1 {
             return Err(AddressEncoderError::InvalidSize);
         };
-        AddressEncoder::try_parse_address(&bytes)
+
+        let network_prefix = (bytes[0] & 0xF0).try_into()?;
+        AddressEncoder::unchecked_parse_address_from_bytes(&bytes)
+            .map(|addr| NetworkAddress {
+                address: addr,
+                network: network_prefix
+            })
+    }
+
+    /// parse address from Base58 encoded string
+    pub fn unchecked_parse_address_from_str(str: &str) -> Result<Address, AddressEncoderError> {
+        let bytes = bs58::decode(str).into_vec()?;
+        AddressEncoder::unchecked_parse_address_from_bytes(&bytes)
     }
 
     /// parse address from Base58 encoded string
-    fn try_parse_address(bytes: &[u8]) -> Result<Address, AddressEncoderError> {
+    pub fn unchecked_parse_address_from_bytes(bytes: &[u8]) -> Result<Address, AddressEncoderError> {
```

Comment:
Note: these "unchecked" functions are all to power the NetworkAddress struct which keeps track of both the network&address in the same struct. Since the NetworkAddress class explicitly takes in a network as a parameter, it's easier if we skip AddresssEncoder network check